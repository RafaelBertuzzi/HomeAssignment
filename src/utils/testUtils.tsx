import { ReactNode } from 'react';
import renderer from 'react-test-renderer';

import { LayoutProvider } from '@components/LayoutProvider';
import { render } from '@testing-library/react-native';

export const rendererWithProviders = (children: ReactNode) => {
  return renderer.create(<LayoutProvider>{children}</LayoutProvider>).toJSON();
};

export const renderWithProviders = (children: ReactNode) => {
  return render(<LayoutProvider>{children}</LayoutProvider>).toJSON();
};
