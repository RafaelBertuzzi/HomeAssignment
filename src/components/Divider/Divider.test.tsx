import { cleanup } from '@testing-library/react-native';
import { rendererWithProviders } from '@utils/testUtils';

import { Divider } from '.';

describe('Component: Divider', () => {
  afterEach(cleanup);

  it('should render divider', () => {
    const json = rendererWithProviders(<Divider />);
    expect(json).toMatchSnapshot();
  });
});
