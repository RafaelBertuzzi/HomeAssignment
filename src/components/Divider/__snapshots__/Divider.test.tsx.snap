// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Divider should render divider 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    aria-orientation="horizontal"
    dataSet={{}}
    style={
      {
        "backgroundColor": "#d4d4d4",
        "height": 1,
        "width": "100%",
      }
    }
    thickness="1"
  />
</RNCSafeAreaProvider>
`;
